### PR TITLE
Enable proguard

### PR DESCRIPTION
**Pull request by l-jonas**
This depends on https://github.com/syncthing/syncthing-lite/pull/48/ to prevent merge conflicts (upon request, I can remove this base)

This reduces the size of an release APK from 6,2 MB to 2,9 MB.

How to test: Run it verify that everything works.

### CODE REVIEW

File: app/proguard-rules.pro [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+-dontwarn org.h2.util.Profiler
+-dontwarn org.h2.util.SourceCompiler
+-dontwarn org.h2.util.SourceCompiler$ClassFileManager
+-dontwarn org.h2.util.SourceCompiler$ClassFileManager$1
+-dontwarn org.h2.util.SourceCompiler$JavaClassObject
+-dontwarn org.h2.util.SourceCompiler$StringJavaFileObject
+-dontwarn org.h2.util.ThreadDeadlockDetector
+-dontwarn org.h2.util.Utils
+-dontwarn org.h2.value.ValueGeometry
+-dontwarn org.h2.value.ValueGeometry$ZVisitor
```
Comment (Nutomic) [inline app/proguard-rules.pro]: Pretty sure you can do `-dontwarn org.h2.*` or something like that (e.g. wildcard for the whole package).

File: app/proguard-debug-rules.pro [hunk 2]
```diff
@@ -0,0 +1 @@
+-keepattributes SourceFile,LineNumberTable
```
Comment (Nutomic) [inline app/proguard-debug-rules.pro]: You also want this for the release build, otherwise it will be impossible to make any sense of crash logs.

**Comment by Nutomic** [comment]
This proguard config is using obfuscation, right? That means we have to keep the mapping.txt for each release, and I don't know if that is possible with F-Droid. I think it's better to use `-dontobfuscate`, so we don't have to deal with all that. 

Did you test if the release build with proguard is working correctly, and how the crash logs look like?

File: app/proguard-rules.pro (hunk 1 above)
Comment (l-jonas) [reply app/proguard-rules.pro]: Yes, that's possible, but I prefer to document the warnings using that. I planned to replace the database by sqlite (which is bundled with Android and the default database engine in many apps) which allows to remove many dontwarn entries.

**Comment by l-jonas** [comment]
An error log (with the keep rules for debugging which I enabled for release builds looks like this):

```
09-28 10:54:51.922  4731  4756 W n*.s*.j*.e.a.a: Table "VERSION" not found; SQL statement:
09-28 10:54:51.922  4731  4756 W n*.s*.j*.e.a.a: SELECT version_number FROM version [42102-196]
09-28 10:54:51.922  4731  4756 W n*.s*.j*.e.a.a:  at org.h2.command.Parser.readTableOrView(Parser.java:5552)
09-28 10:54:51.922  4731  4756 W n*.s*.j*.e.a.a:  at org.h2.command.Parser.readTableFilter(Parser.java:1266)
09-28 10:54:51.922  4731  4756 W n*.s*.j*.e.a.a:  at org.h2.command.Parser.parseSelectSimpleFromPart(Parser.java:1946)
09-28 10:54:51.922  4731  4756 W n*.s*.j*.e.a.a:  at org.h2.command.Parser.parseSelectSimple(Parser.java:2095)
09-28 10:54:51.922  4731  4756 W n*.s*.j*.e.a.a:  at org.h2.command.Parser.parseSelectSub(Parser.java:1940)
09-28 10:54:51.922  4731  4756 W n*.s*.j*.e.a.a:  at org.h2.command.Parser.parseSelectUnion(Parser.java:1755)
09-28 10:54:51.922  4731  4756 W n*.s*.j*.e.a.a:  at org.h2.command.Parser.parseSelect(Parser.java:1743)
09-28 10:54:51.922  4731  4756 W n*.s*.j*.e.a.a:  at org.h2.command.Parser.parsePrepared(Parser.java:449)
09-28 10:54:51.922  4731  4756 W n*.s*.j*.e.a.a:  at org.h2.command.Parser.parse(Parser.java:321)
09-28 10:54:51.922  4731  4756 W n*.s*.j*.e.a.a:  at org.h2.command.Parser.parse(Parser.java:293)
09-28 10:54:51.922  4731  4756 W n*.s*.j*.e.a.a:  at org.h2.command.Parser.prepareCommand(Parser.java:258)
09-28 10:54:51.922  4731  4756 W n*.s*.j*.e.a.a:  at org.h2.engine.Session.prepareLocal(Session.java:578)
09-28 10:54:51.922  4731  4756 W n*.s*.j*.e.a.a:  at org.h2.engine.Session.prepareCommand(Session.java:519)
09-28 10:54:51.922  4731  4756 W n*.s*.j*.e.a.a:  at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1204)
09-28 10:54:51.922  4731  4756 W n*.s*.j*.e.a.a:  at org.h2.jdbc.JdbcPreparedStatement.<init>(JdbcPreparedStatement.java:73)
09-28 10:54:51.922  4731  4756 W n*.s*.j*.e.a.a:  at org.h2.jdbc.JdbcConnection.prepareStatement(JdbcConnection.java:288)
09-28 10:54:51.922  4731  4756 W n*.s*.j*.e.a.a:  at com.b.a.b.j.prepareStatement(ProxyConnection.java:308)
09-28 10:54:51.922  4731  4756 W n*.s*.j*.e.a.a:  at com.b.a.b.c.prepareStatement(HikariProxyConnection.java)
09-28 10:54:51.922  4731  4756 W n*.s*.j*.e.a.a:  at net.syncthing.java.e.a.a.e(SqlRepository.kt:89)
09-28 10:54:51.922  4731  4756 W n*.s*.j*.e.a.a:  at net.syncthing.java.e.a.a.<init>(SqlRepository.kt:62)
09-28 10:54:51.922  4731  4756 W n*.s*.j*.e.a.a:  at net.syncthing.java.b.a.<init>(SyncthingClient.kt:42)
09-28 10:54:51.922  4731  4756 W n*.s*.j*.e.a.a:  at net.syncthing.lite.library.d.<init>(LibraryInstance.kt:43)
09-28 10:54:51.922  4731  4756 W n*.s*.j*.e.a.a:  at net.syncthing.lite.library.a$b.b(DefaultLibraryManager.kt:42)
09-28 10:54:51.922  4731  4756 W n*.s*.j*.e.a.a:  at net.syncthing.lite.library.a$b.a(DefaultLibraryManager.kt:11)
09-28 10:54:51.922  4731  4756 W n*.s*.j*.e.a.a:  at net.syncthing.lite.library.e$d.run(LibraryManager.kt:45)
09-28 10:54:51.922  4731  4756 W n*.s*.j*.e.a.a:  at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:428)
09-28 10:54:51.922  4731  4756 W n*.s*.j*.e.a.a:  at java.util.concurrent.FutureTask.run(FutureTask.java:237)
09-28 10:54:51.922  4731  4756 W n*.s*.j*.e.a.a:  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
09-28 10:54:51.922  4731  4756 W n*.s*.j*.e.a.a:  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
09-28 10:54:51.922  4731  4756 W n*.s*.j*.e.a.a:  at java.lang.Thread.run(Thread.java:761)
09-28 10:54:51.922  4731  4756 I n*.s*.j*.e.a.a: init db
09-28 10:54:52.030  4731  4756 I n*.s*.j*.e.a.a: database initialized
09-28 10:54:52.046  4731  4764 I n*.s*.j*.c.a.b: Opened udp socket ::/:::21027
```

This file names and line numbers at the end are usable, the package names not. I think that this is good enough so that the mapping.txt is not required. I tested the release build and it worked.

File: app/proguard-rules.pro (hunk 1 above)
Comment (Nutomic) [reply app/proguard-rules.pro]: Yes that definitely makes sense I think.

File: app/build.gradle [hunk 3]
```diff
@@ -35,10 +35,13 @@ android {
     }
     buildTypes {
         debug {
-
+            minifyEnabled true
+            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
```
Comment (Nutomic) [inline app/build.gradle]: This is probably a bad idea, because proguard slows down builds a lot.

File: app/proguard-rules.pro (hunk 1 above)
Comment (Nutomic) [reply app/proguard-rules.pro]: I'd prefer to keep it simple, and having that many lines isn't really simple. But if you prefer, we can go with this.

And yes, using the native Android database is probably a good idea.

File: app/build.gradle (hunk 3 above)
Comment (l-jonas) [reply app/build.gradle]: This does not cause any performance issues for me. But when using android instant run (the play button of Android Studio), this causes some random crashs due to missing classes (which don't occur when generating a debuggable APK and installing that one). On the other hand, it's better to test with proguard to find bugs caused by it, but as it works now, I think it's better to disable it.

Comment (Nutomic) [reply app/build.gradle]: We should definitely test a release build of the app before every release. But I don't think it's a good idea to do that every time you debug something.

**Pull request merged**